editor/src/app/app-state/app-state.service.ts

```diff
@@ -30,6 +31,52 @@ export class AppStateService {
     private router: Router) {
   }
 
+  showLoading() {
+    this.store.dispatch(new AppShowLoading());
+  }
+
+  hideLoading() {
+    this.store.dispatch(new AppHideLoading());
+  }
+
+  sync(urlName: string, data: any, method?: string) {
+
+    return this.store.select(state => state.app)
+      .pipe(
+        filter(appState => !!appState.user.token && appState.urls[urlName]),  // Make sure user is logged in
+        take(1),
+        map(state => {
+          this.showLoading();
+          return fetch(
```

Comment:
please use Angular HTTP here

Reply:
Already tried but give up. Couldn't figure out how to pass `method`.

Reply:
This would probably be best to be split into several methods since they do different things when we use different HTTP methods.

However, if you wish to reimplement our old sync one to one. You can use the request method of Angular HTTP client: https://angular.io/api/common/http/HttpClient#request.
E.G.:
```ts
this.http.request<any>('GET', url, {
    body: data,
    headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
        'X-Authorization': 'Bearer ' + state.user.token
    },
    ...
});
```